corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js

```diff
@@ -183,10 +183,66 @@ function Form(json) {
     Container.call(self, json);
     self.submitText = ko.observable('Submit');
 
+    self.currentIndex = ko.observable(0);
+    self.atLastIndex = ko.observable(false);
+
+    var _updateIndexCallback = function (ix, isAtLastIndex) {
+        self.currentIndex(ix);
+        self.atLastIndex(isAtLastIndex);
+    };
+
+    self.showInFormNavigation = ko.observable(
+        self.displayOptions.oneQuestionPerScreen !== undefined
+        && self.displayOptions.oneQuestionPerScreen() === true
+    );
+
+    self.isCurrentRequiredSatisfied = ko.computed(function () {
+        if (!self.showInFormNavigation()) return true;
+
+        return _.every(self.children(), function (q) {
+            return (q.answer() === null && !q.required()) || q.answer() !== null;
```

Comment:
nit: in most of the code i use `Const.NO_ANSWER` so it's more explicit. there's a subtle difference between `null` and `undefined` for the answer and i think using the constants make it more clear